mpas_analysis/shared/plot/plotting.py

```diff
@@ -638,6 +638,8 @@ def plot_polar_projection_comparison(
         titleFontSize=None,
         figsize=None,
         dpi=None,
+        lineWidth=0.5,
+        lineColor='black',
```

Comment:
Surely the default for both of these should be `None`?

Reply:
I don't know.  My initial reason for adding these was for consistency with other code in plotting.py (it's done this way in plot_global_comparison, I think, and the new plot_vertical_section_comparison inherited it as well).  If lineWidth is None when passed to contour(), a Matplotlib default is used;  there is also the opportunity to intercept a None before calling contour(), and to then check for the presence of a lineWidth parameter from the config file.  In any case, it does seem to be the case that different plotting.py functions require different defaults (the default lineWidth for single panel plots is too thick for the three-panel plots, for example).  Regarding lineColor, if None when passed to contour for the colors parameter, Matplotlib has its own default behaviour (again, we could intercept the None and check the config file).  It seems to me that black is a reasonable default for contour  lines, and I think this is done elsewhere, but as we discussed before, there is sometimes not consistency in plotting.py in regard to such matters.  These sort of issues are probably all encompassed by the plotting.py cleanup task that you opened today as issue #385.

Reply:
Fill me in on your philosophy regarding defaults for keyword parameters, if I am missing something.

Reply:
I see.  Okay, first the general philosophy: these plotting functions really shouldn't try to guess what a reasonable line width and color is.  These should either come from a config option or default to the matplotlib default if they aren't explicitly specified.  Then, there's the argument specific to these transect plots: I think the default should be no contours unless the calling code explicitly asks for them.  Does that make sense?

Reply:
And I can completely imagine this is the kind of thing that would get inherited from our other plotting functions, none of which has a particularly coherent philosophy or coding style, I'm afraid.  Fortunately, you'll help save us from ourselves when you get to addressing #385 ;-)